src/test/regress/citus_tests/join_gen/tests/diff-checker.py

```diff
@@ -0,0 +1,121 @@
+import os
+import re
+import subprocess
+import sys
+
+
+def createPatternForFailedQueryBlock(acceptableErrors):
+    totalAcceptableOrders = len(acceptableErrors)
+
+    failedQueryBlockPattern = "-- queryId: [0-9]+\n.*\npsql:.*(?:"
+    for acceptableErrorIdx in range(totalAcceptableOrders):
+        failedQueryBlockPattern += acceptableErrors[acceptableErrorIdx]
+        if acceptableErrorIdx < totalAcceptableOrders - 1:
+            failedQueryBlockPattern += "|"
+    failedQueryBlockPattern += ")"
+
+    return failedQueryBlockPattern
+
+
+def findFailedQueriesFromFile(queryOutFile, acceptableErrors):
+    failedQueryIds = []
+    outFileContent = ""
+    failedQueryBlockPattern = createPatternForFailedQueryBlock(acceptableErrors)
+    queryIdPattern = "queryId: ([0-9]+)"
+    with open(queryOutFile, "r") as f:
+        outFileContent = f.read()
+        failedQueryContents = re.findall(failedQueryBlockPattern, outFileContent)
+        failedQueryIds = [
+            re.search(queryIdPattern, failedQueryContent)[1]
+            for failedQueryContent in failedQueryContents
+        ]
+
+    return failedQueryIds
+
+
+def removeFailedQueryOutputFromFile(outFile, failedQueryIds):
+    if len(failedQueryIds) == 0:
+        return
+
+    distOutFileContentAsLines = []
+    with open(outFile, "r") as f:
+        distOutFileContentAsLines = f.readlines()
+
+    with open(outFile, "w") as f:
+        clear = False
+        nextIdx = 0
+        nextQueryIdToDelete = failedQueryIds[nextIdx]
+        queryIdPattern = "queryId: ([0-9]+)"
+
+        for line in distOutFileContentAsLines:
+            matched = re.search(queryIdPattern, line)
+            # founded line which contains query id
+            if matched:
+                # query id matches with the next failed query's id
+                if nextQueryIdToDelete == matched[1]:
+                    # clear lines until we find succesfull query
+                    clear = True
+                    nextIdx += 1
+                    if nextIdx < len(failedQueryIds):
+                        nextQueryIdToDelete = failedQueryIds[nextIdx]
+                else:
+                    # we found successfull query
+                    clear = False
+
+            if not clear:
+                f.write(line)
+    return
+
+
+def removeFailedQueryOutputFromFiles(distQueryOutFile, localQueryOutFile):
+    # remove the failed distributed query from both local and distributed query files to prevent diff for acceptable errors
+    acceptableErrors = [
+        "ERROR:  complex joins are only supported when all distributed tables are co-located and joined on their distribution columns",
```

Comment:
Can you add a comment why are these failures happening? Is that due to https://github.com/citusdata/citus/issues/6653?